### ISSUE TITLE
Crash in SFCollectIosDeviceProperties

### ISSUE THREAD

**Issue by edenman** [SiftScience/sift-ios#67]
Just got this crash in the wild.
Sift 0.9.8.  iOS 10.3.2. iPhone7,2
```
NSInvalidArgumentException: Invalid number value (NaN) in JSON write

0  CoreFoundation          ___exceptionPreprocess
1  libobjc.A.dylib         _objc_exception_throw
2  CoreFoundation          +[NSException raise:format:]
3  Foundation              __writeJSONNumber
4  Foundation              __writeJSONValue
5  Foundation              ____writeJSONObject_block_invoke
6  CoreFoundation          ___65-[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:]_block_invoke
7  CoreFoundation          -[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:]
8  Foundation              __writeJSONObject
9  Foundation              __writeJSONValue
10 Foundation              ____writeJSONObject_block_invoke
11 CoreFoundation          ___65-[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:]_block_invoke
12 CoreFoundation          -[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:]
13 Foundation              __writeJSONObject
14 Foundation              __writeJSONValue
15 Foundation              ____writeJSONObject_block_invoke
16 CoreFoundation          ___65-[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:]_block_invoke
17 CoreFoundation          -[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:]
18 Foundation              __writeJSONObject
19 Foundation              __writeJSONValue
20 Foundation              ____writeJSONArray_block_invoke
21 CoreFoundation          ___53-[__NSArrayM enumerateObjectsWithOptions:usingBlock:]_block_invoke
22 CoreFoundation          -[__NSArrayM enumerateObjectsWithOptions:usingBlock:]
23 Foundation              __writeJSONArray
24 Foundation              __writeJSONValue
25 Foundation              ____writeJSONObject_block_invoke
26 CoreFoundation          -[__NSSingleEntryDictionaryI enumerateKeysAndObjectsWithOptions:usingBlock:]
27 Foundation              __writeJSONObject
28 Foundation              __writeJSONValue
29 Foundation              -[_NSJSONWriter dataWithRootObject:options:error:]
30 Foundation              +[NSJSONSerialization dataWithJSONObject:options:error:]
31 Sift                    0x7490 (-0x1011d0000 + 4313674896)
32 Sift                    SFCollectIosDeviceProperties
33 libdispatch.dylib       __dispatch_call_block_and_release
34 libdispatch.dylib       __dispatch_client_callout
35 libdispatch.dylib       __dispatch_queue_serial_drain
36 libdispatch.dylib       __dispatch_queue_invoke
37 libdispatch.dylib       __dispatch_queue_override_invoke
38 libdispatch.dylib       __dispatch_root_queue_drain
39 libdispatch.dylib       __dispatch_worker_thread3
40 libsystem_pthread.dylib __pthread_wqthread
41 libsystem_pthread.dylib _start_wqthread
```

**Comment by garylee1** [comment]
@edenman we fixed this with v0.9.9 https://github.com/SiftScience/sift-ios/releases/tag/v0.9.9

Though I probably recommend upgrading to v0.9.10 which prepares our SDK for iOS 11

**Comment by farmer-dhi** [comment]
Had two crashes in this same file today. Using SDK version 2.2.4

Here is the stack trace our logging system gave us

         Crashed: com.apple.main-thread
0  libsystem_platform.dylib       0xf54 _platform_strstr + 52
1  MyApp                           0x4ba054 SFCollectIosDeviceProperties + 250 (SiftIosDeviceProperties.m:250)
2  MyApp                           0x4ba630 -[SiftIosDevicePropertiesCollector collect] + 67 (SiftIosDevicePropertiesCollector.m:67)
3  CoreFoundation                 0x14f80 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 148
4  CoreFoundation                 0x17fb0 ___CFXRegistrationPost_block_invoke + 92
5  CoreFoundation                 0x17ef4 _CFXRegistrationPost + 436
6  CoreFoundation                 0x15a4c _CFXNotificationPost + 736
7  Foundation                     0x127ac -[NSNotificationCenter postNotificationName:object:userInfo:] + 92
8  UIKitCore                      0x19a078 -[UIApplication _sendWillEnterForegroundCallbacks] + 212
9  UIKitCore                      0x170e50 __101-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]_block_invoke_2 + 1280
10 UIKitCore                      0x16c46c _UIScenePerformActionsWithLifecycleActionMask + 112
11 UIKitCore                      0x170870 __101-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]_block_invoke + 252
12 UIKitCore                      0x1705e0 -[_UISceneLifecycleMultiplexer _performBlock:withApplicationOfDeactivationReasons:fromReasons:] + 212
13 UIKitCore                      0x1702c8 -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 608
14 UIKitCore                      0x16ff30 -[_UISceneLifecycleMultiplexer uiScene:transitionedFromState:withTransitionContext:] + 244
15 UIKitCore                      0x16ca5c __186-[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block_invoke + 148
16 UIKitCore                      0x16d078 +[BSAnimationSettings(UIKit) tryAnimatingWithSettings:fromCurrentState:actions:completion:] + 736
17 UIKitCore                      0x16c99c _UISceneSettingsDiffActionPerformChangesWithTransitionContextAndCompletion + 224
18 UIKitCore                      0x16c64c -[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:] + 316
19 UIKitCore                      0x16c224 __64-[UIScene scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke.229 + 616
20 UIKitCore                      0x16bb7c -[UIScene _emitSceneSettingsUpdateResponseForCompletion:afterSceneUpdateWork:] + 208
21 UIKitCore                      0x16b9f4 -[UIScene scene:didUpdateWithDiff:transitionContext:completion:] + 244
22 UIKitCore                      0x2c43e0 -[UIApplicationSceneClientAgent scene:handleEvent:withCompletion:] + 336
23 FrontBoardServices             0x24094 __76-[FBSScene updater:didUpdateSettings:withDiff:transitionContext:completion:]_block_invoke.146 + 252
24 FrontBoardServices             0x1f6fc -[FBSScene _callOutQueue_coalesceClientSettingsUpdates:] + 68
25 FrontBoardServices             0x23dc0 -[FBSScene updater:didUpdateSettings:withDiff:transitionContext:completion:] + 712
26 FrontBoardServices             0x26ca0 __94-[FBSWorkspaceScenesClient _queue_updateScene:withSettings:diff:transitionContext:completion:]_block_invoke_2 + 148
27 FrontBoardServices             0x1ed38 -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 168
28 FrontBoardServices             0x26bcc __94-[FBSWorkspaceScenesClient _queue_updateScene:withSettings:diff:transitionContext:completion:]_block_invoke.cold.1 + 252
29 FrontBoardServices             0x26a40 __94-[FBSWorkspaceScenesClient _queue_updateScene:withSettings:diff:transitionContext:completion:]_block_invoke + 184
30 libdispatch.dylib              0x1b584 _dispatch_client_callout + 16
31 libdispatch.dylib              0x6ab0 _dispatch_block_invoke_direct + 284
32 FrontBoardServices             0x1e8ac __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 52
33 FrontBoardServices             0x1e748 -[FBSMainRunLoopSerialQueue _targetQueue_performNextIfPossible] + 240
34 FrontBoardServices             0x1e7b0 -[FBSMainRunLoopSerialQueue _performNextFromRunLoopSource] + 28
35 CoreFoundation                 0xf92c __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
36 CoreFoundation                 0xf744 __CFRunLoopDoSource0 + 172
37 CoreFoundation                 0xf5a0 __CFRunLoopDoSources0 + 232
38 CoreFoundation                 0xff20 __CFRunLoopRun + 840
39 CoreFoundation                 0x11adc CFRunLoopRunSpecific + 572
40 GraphicsServices               0x1454 GSEventRunModal + 168
41 UIKitCore                      0x135274 -[UIApplication _run] + 816
42 UIKitCore                      0x100a28 UIApplicationMain + 336
43 MyApp                           0x40a8 main + 11 (main.swift:11)
44 ???                            0x1c9085f08 (Missing)
